### PR TITLE
Mmt 1666 create update function for mmt proposal mode

### PR DESCRIPTION
**Pull request by htranho**

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -121,7 +121,13 @@
 
   scope module: :proposal do
     resource :manage_collection_proposals, only: :show
-    resources :collection_draft_proposals, only: [:new, :create, :edit, :show]
+    resources :collection_draft_proposals, controller: 'collection_draft_proposals', draft_type: 'CollectionDraftProposal', as: 'collection_draft_proposals' do
+      member do
+        get 'edit', path: 'edit(/:form)'
+        get 'download'
```
Comment (htranho) [inline config/routes.rb]: remove this

File: config/routes.rb [hunk 2]
```diff
@@ -121,7 +121,13 @@
...
+      end
+    end
+    get 'subregion_options' => 'collection_draft_proposals#subregion_options'
```
Comment (htranho) [inline config/routes.rb]: verify if we need it

File: spec/features/collection_draft_proposals/collection_draft_proposal_update_spec.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      before do
+        fill_in 'Short Name', with: '123'
+        fill_in 'Abstract', with: 'collection abstract'
+        within '.nav-top' do
+          click_on 'Save'
+        end
+      end
+
+      it 'displays a confirmation message' do
+        save_screenshot '/tmp/s.png'
```
Comment (htranho) [inline spec/features/collection_draft_proposals/collection_draft_proposal_update_spec.rb]: remove this

File: spec/factories/valid_collection_draft_proposal_factories.rb [hunk 4]
```diff
@@ -0,0 +1,893 @@
+require 'faker'
```
Comment (ryanmiller-1) [inline spec/factories/valid_collection_draft_proposal_factories.rb]: https://github.com/nasa/mmt/pull/382/files#diff-a36a537bfe0e4dbbc278308dc21778cd
https://github.com/nasa/mmt/pull/382/files#diff-04aebb6b804325ff69280a86ac31c27f

Comment (ryanmiller-1) [reply spec/factories/valid_collection_draft_proposal_factories.rb]: In summary, I stuffed all of the reusable chunks of the factory into a function and then merged anything that we might want to change in afterwards.

File: app/views/collection_drafts/progress/_preview_progress.html.erb [hunk 5]
```diff
@@ -3,9 +3,9 @@
 
     <h3 class="inline-header">Metadata Fields</h3><a href="#progress-legend" id="legend" class="display-modal"><i class="eui-icon eui-fa-info-circle"></i> <span class="is-invisible">Progress Legend</span></a>
     <ul class="progress-indicator">
-      <% CollectionDraft.forms.each do |form_name| %>
+      <% @resource_class.forms.each do |form_name| %>
       <li id="<%= form_name.dasherize %>">
-        <%= render partial: 'collection_drafts/progress/form_progress', locals: { draft: draft, form_name: form_name, metadata_errors: metadata_errors } %>
+        <%= render partial: 'collection_drafts/progress/form_progress', locals: { unpublished_record: unpublished_record, form_name: form_name, metadata_errors: metadata_errors } %>
```
Comment (htranho) [inline app/views/collection_drafts/progress/_preview_progress.html.erb]: change to unpublished_resource

File: config/routes.rb (hunk 2 above)
Comment (htranho) [reply config/routes.rb]: this is needed for the subregion (state) selection for address.

**Comment by ryanmiller-1** [comment]
I think the collection drafts and all of their derivatives should be able to use the same base file in terms of factory data (at least for valid data, I did not check what makes the invalid data invalid).  I think that the file should have a more generic name, long term.  I am not sure that needs to hold up this ticket if it gets fixed in a later ticket of the feature branch though.

**Pull request merged**